gspread/auth.py

```diff
@@ -189,7 +189,7 @@ def oauth(
     authorized_user_filename = Path(authorized_user_filename)
     creds = load_credentials(filename=authorized_user_filename)
 
-    if not type(creds) is Credentials:
+    if not isinstance(creds, OAuthCredentials):
```

Comment:
very good catch ! 

Here we should compare to the base class `Credentials` because the function `load_credentials` above can return an instance of type `google.auth.external_account_authorized_user.Credentials` or `google.oauth2.credentials.Credentials`

Reply:
sure, makes sense.

I chose `OAuthCredentials` because this is under the `oauth()` method. But, as I said, I do not know oauth well, so I have changed.